### PR TITLE
Add support for multi-line string encoding

### PR DESCRIPTION
**Pull request by k1LoW**
Hi @goccy !

I added support for multi-line string encoding similar to go-yaml/yaml.

**go-yaml/yaml encoding result:**

https://play.golang.org/p/mEGR5B3JpNY

**github.com/goccy/go-yaml v1.3.0 encoding result:**

https://play.golang.org/p/EDXASMpmbiV

Thanks.

### CODE REVIEW

File: ast/ast.go [hunk 1]
```diff
@@ -420,14 +420,26 @@ func (n *StringNode) GetValue() interface{} {
...
+// String string value to text with quote or literal header if required
 func (n *StringNode) String() string {
 	switch n.Token.Type {
 	case token.SingleQuoteType:
 		return fmt.Sprintf(`'%s'`, n.Value)
 	case token.DoubleQuoteType:
 		return fmt.Sprintf(`"%s"`, n.Value)
 	}
+
+	if strings.Contains(n.Value, "\n") {
```
Comment (goccy) [inline ast/ast.go]: Looks good !!
But we may have to consider CRLF for windows .

Comment (k1LoW) [reply ast/ast.go]: Yeah, you right.

**Comment by k1LoW** [comment]
@goccy 
I tried fixes, but not perfect.

Because it seems to be necessary to detect `the line break character of YAML file` to detect `the line break character of the string content`. ( see https://yaml.org/spec/1.2/spec.html#id2774608 ).

---

For example, [ast.LiteralNode use only `"\n"`](https://github.com/goccy/go-yaml/blob/fee8ac5323073d5fcc780ae9cc9dbdf8ac9ad428/ast/ast.go#L465). 

One way is to make ast.StringNode the same implementation as ast.LiteralNode.

**Comment by k1LoW** [comment]
go-yaml/yaml does not seems to detect line break character.
https://play.golang.org/p/ZXnPJoVp0kT

**Comment by goccy** [comment]
@k1LoW 
Thanks for trying to fix it !
Merge and improve this problem if needed :)

LGTM !

**Pull request merged**